internal/api/ui/login/mfa_verify_otp_handler.go

```diff
@@ -0,0 +1,106 @@
+package login
+
+import (
+	"fmt"
+	"net/http"
+
+	http_mw "github.com/zitadel/zitadel/internal/api/http/middleware"
+	"github.com/zitadel/zitadel/internal/domain"
+)
+
+const (
+	tmplOTPVerification   = "otpverification"
+	querySelectedProvider = "selectedProvider"
+)
+
+type mfaOTPData struct {
+	userData
+	MFAProviders     []domain.MFAType
+	SelectedProvider domain.MFAType
+}
+
+type mfaOTPFormData struct {
+	Resend           bool           `schema:"resend"`
+	Code             string         `schema:"code"`
+	SelectedProvider domain.MFAType `schema:"selectedProvider"`
+	Provider         domain.MFAType `schema:"provider"`
+}
+
+func OTPLink(origin, authRequestID, code string, provider domain.MFAType) string {
+	return fmt.Sprintf("%s%s?%s=%s&%s=%s&%s=%d", externalLink(origin), EndpointMFAOTPVerify, QueryAuthRequestID, authRequestID, queryCode, code, querySelectedProvider, provider)
+}
+
+// renderOTPVerification renders the OTP verification for SMS and Email based on the passed MFAType.
+// It will send a new code to either phone or email first.
+func (l *Login) handleOTPVerification(w http.ResponseWriter, r *http.Request, authReq *domain.AuthRequest, providers []domain.MFAType, selectedProvider domain.MFAType, err error) {
+	if err != nil {
+		l.renderOTPVerification(w, r, authReq, providers, selectedProvider, err)
+		return
+	}
+	userAgentID, _ := http_mw.UserAgentIDFromCtx(r.Context())
+	sendCode := l.authRepo.SendMFAOTPSMS
```

Comment:
Is the deault always SMS?

Reply:
it's just one or the other... should i check for both?

Reply:
I would do so yes, as it is not really that much more work